migration-tools/migrate-rosdistro.py

```diff
@@ -221,8 +221,19 @@ def write_tracks_file(tracks, commit_msg=None):
         # Bloom will not run with multiple remotes.
         subprocess.check_call(['git', 'remote', 'remove', 'oldorigin'])
         subprocess.check_call(['git', 'bloom-release', '--non-interactive', '--release-increment', release_inc, '--unsafe', args.dest], stdin=subprocess.DEVNULL, env=os.environ)
-        subprocess.check_call(['git', 'push', 'origin', '--all', '--force'])
-        subprocess.check_call(['git', 'push', 'origin', '--tags', '--force'])
+
+        # If something goes wrong, use dry run pushes to ensure we don't
+        # destroy any data in the repository.
+        # 1. Ensure we're not overwriting tags
+        # 2. Push `upstream` branch. This is the ONLY branch we should ever
+        #    force-push, and is an intermediate staging branch anyway.
+        # 3. Ensure we're not overwriting any branches
+        # 4. Push the branches and tags
+        subprocess.check_call(['git', 'push', 'origin', '--tags', '--dry-run'])
+        subprocess.check_call(['git', 'push', 'origin', 'upstream', '--force'])
```

Comment:
There are situations where `upstream/*` tags may also technically be changed between releases which would result in an inconsistent state between the current upstream state and the previous tag but the only real-world examples I know of were in the early ROS 2 days with fastrtps and thankfully it doesn't happen any more.

Reply:
I don't think this requires action right away. It's a hint in case this change creates additional unexpected failures pushing other upstream tags.